service/history/taskProcessor.go

```diff
@@ -211,6 +211,12 @@ FilterLoop:
 				scope.RecordTimer(metrics.TaskAttemptTimer, time.Duration(task.attempt))
 				task.logger.Error("Critical error processing task, retrying.",
 					tag.Error(err), tag.OperationCritical, tag.TaskType(task.task.GetTaskType()))
+				if isStickyTaskConditionError(err) {
+					// sticky task could end up into endless loop in rare cases and
+					// cause worker keep getting decision timeout unless restart.
+					// return nil here to break the endless loop
+					return nil
+				}
```

Comment:
Seems like the logic belongs to `handleTaskError` func? task.attempt should be available there. 

Can you also update the same logic for new task processing logic in `service/history/task/task.go`?

Reply:
Moved and added